src/main/java/unicash/model/transaction/Location.java

```diff
@@ -9,13 +9,13 @@
 public class Location {
     public static final String MESSAGE_CONSTRAINTS =
             "Locations should only contain alphanumeric characters, spaces, (, ), _, -, #, &, ., and ',', "
-                    + "and it should not be blank";
+                    + "up to 500 characters and it should not be blank";
 
     /*
      * The first character of the location must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
-    public static final String VALIDATION_REGEX = "[\\p{Alnum}_&#.,()-][\\p{Alnum} _&#.,()-]*";
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}_&#.,()-][\\p{Alnum} _&#.,()-]{0,499}";
```

Comment:
```suggestion
    public static final String VALIDATION_REGEX = "[\\p{Alnum}_&#.,()-][\\p{Alnum} _&#.,()-@]{0,499}";
```

Reason is, users might wanna input sub-locations i.e. `McDonalds @ Clementi`

Reply:
Believe the `@` symbol must come before the `-` symbol